### PR TITLE
support scripts for host c++compiling

### PR DESCRIPTION
**Pull request by alan-stokes**
title

depends upon 
- [ ] FEC https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/303
- [x] spinnaker tools: https://github.com/SpiNNakerManchester/spinnaker_tools/pull/72

### CODE REVIEW

**Comment by dkfellows** [comment]
What's the value in having these changes?

File: automatic_host_make.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+cd SpiNNFrontEndCommon/cpp_common/
+make clean
+make install
+cd ../..
+echo "completed host binary compilation"
```
Comment (dkfellows) [inline automatic_host_make.sh]: Lines 1–4 should be this:
```
make -C SpiNNFrontEndCommon/cpp_common clean install
```

File: automatic_spinnaker_make.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# This script assumes it is run from the directory holding all github projects in parellel
+# sh SupportScripts/automatic_make.sh
+
```
Comment (dkfellows) [inline automatic_spinnaker_make.sh]: You should add this to the script at this point:
```
set -e
```
It makes the shell do automatic error handling; it exits on any error and that's probably what you want for this whole script. (It makes errors work a lot like exceptions.)

File: automatic_spinnaker_make.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+cd spinnaker_tools
+source $PWD/setup
+make clean
+make || exit $?
+cd ..
```
Comment (dkfellows) [inline automatic_spinnaker_make.sh]: Lines 4–8 should be this:
```
cd spinnaker_tools; source setup; cd ..
make -C spinnaker_tools clean all
```

File: automatic_spinnaker_make.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+cd spinn_common
+make clean
+make || exit $?
+make install
+cd ..
```
Comment (dkfellows) [inline automatic_spinnaker_make.sh]: Lines 9–13 should be:
```
make -C spinn_common clean all install
```

File: automatic_spinnaker_make.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+cd SpiNNFrontEndCommon/c_common/
+cd front_end_common_lib/
+make install-clean
+cd ..
+make clean
+make || exit $?
+make install
+cd ../..
```
Comment (dkfellows) [inline automatic_spinnaker_make.sh]: Lines 14–21 should be:
```
make -C SpiNNFrontEndCommon/c_common/front_end_common_lib install-clean
make -C SpiNNFrontEndCommon/c_common clean all install
```

File: automatic_spinnaker_make.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+cd sPyNNaker/neural_modelling/
+make clean
+make || exit $?
+source $PWD/setup
```
Comment (dkfellows) [inline automatic_spinnaker_make.sh]: Lines 22-25 should be:
```
source sPyNNaker/neural_modelling/setup.bash
make -C sPyNNaker/neural_modelling clean all
```
(The above depends on my proposed rename of `setup` which also addresses the really big bugs in it.)

File: automatic_spinnaker_make.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+cd ../../SpiNNakerGraphFrontEnd/spinnaker_graph_front_end/examples/
+make clean
+make || exit $?
```
Comment (dkfellows) [inline automatic_spinnaker_make.sh]: Lines 26–28 should be:
```
make -C SpiNNakerGraphFrontEnd/spinnaker_graph_front_end/examples clean all
```

File: automatic_host_make.sh (hunk 1 above)
Comment (dkfellows) [reply automatic_host_make.sh]: Also doing `set -e` beforehand would be wise; it makes things much less inclined to go wrong silently.

**Comment by alan-stokes** [comment]
as all these issues are to due to how the scripts works in general, and not on the splitting of host and machine. Ill hand it over to ya to fix as you see fit.

**Comment by dkfellows** [comment]
SpiNNakerManchester/sPyNNaker#493 will break this PR

**Comment by rowleya** [comment]
I think this is now obsolete

**Pull request closed without merging**